# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureClient.cpp
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d gfx/layers/client/TextureClient.cpp
--- a/gfx/layers/client/TextureClient.cpp	Thu Jun 11 06:36:55 2020 +0000
+++ b/gfx/layers/client/TextureClient.cpp	Thu Jun 11 06:37:35 2020 +0000
@@ -324,6 +324,15 @@
   return TextureType::Unknown;
 }
 
+TextureType PreferredCanvasTextureType(
+    const KnowsCompositor& aKnowsCompositor) {
+  const auto layersBackend = aKnowsCompositor.GetCompositorBackendType();
+  const auto moz2DBackend =
+      BackendTypeForBackendSelector(layersBackend, BackendSelector::Canvas);
+  return GetTextureType(gfx::SurfaceFormat::R8G8B8A8, {1, 1}, layersBackend,
+                        moz2DBackend, 2, TextureAllocationFlags::ALLOC_DEFAULT);
+}
+
 static bool ShouldRemoteTextureType(TextureType aTextureType,
                                     BackendSelector aSelector) {
   if (!XRE_IsContentProcess()) {